In-component drawer

luisb90 opened the pull request:
This PR updates the drawer component to give it the ability to open a drawer from within a parent container. an `isRoot` and `parentContainer` options have been added to the drawer options. When `isRoot` is set to false, you can supply an html element as a parent container, and the injection service will use the Portal api to inject the drawer as a child of that parent element. As part of this PR i also cleaned up code that wasnt being used properly.

<img width="864" alt="Screen Shot 2020-01-21 at 7 35 29 AM" src="https://user-images.githubusercontent.com/5652524/72809533-6d168080-3c21-11ea-8e8f-816535a0e61e.png">

<img width="855" alt="Screen Shot 2020-01-21 at 7 36 31 AM" src="https://user-images.githubusercontent.com/5652524/72809534-6d168080-3c21-11ea-87fe-395213ae1654.png">

luisb90 commented on projects/swimlane/ngx-ui/src/lib/components/drawer/drawer.component.ts:
```diff
@@ -90,48 +93,18 @@ export class DrawerComponent implements OnDestroy {
-      this.heightSize = height;
-      this.widthSize = width;
-      this.transform = transform;
-    });
```
I removed this code because during testing, I noticed that this code that was calculating the transform was never updating the element's transform property. The transform set by the drawer animation always took precedence and would not change even after subsequent calls to `setDimensions`. I also noticed that the code was calculating exact px sizes, even though the size property of the element is in percentages relative to 100%. It worked in that way when the width or height were just set to the specified percentage value. So I was able to greatly simplify this code.

aacebo commented on projects/swimlane/ngx-ui/src/lib/components/drawer/drawer.component.ts:
```diff
@@ -81,7 +80,7 @@ export class DrawerComponent implements OnInit, OnDestroy {
   widthSize: string | number;
   heightSize: string | number;
-  position: 'fixed' | 'initial' = 'fixed';
+  position: 'fixed' | 'absolute' = 'absolute';
```
can we change this to an enum instead of using magic strings?

luisb90 replied:
enumed it up

aacebo commented on src/app/dialogs/drawer-page/drawer-page.component.html:
```diff
@@ -162,8 +162,8 @@ <h1>Attack Type: Malware</h1>
+    <button type="button" (click)="openDrawer('left', undefined, true, editTmpl2, false, parentContainer)">open drawer time</button>
       <div [style.width.%]="100" [style.height.px]="500" [style.background-color]="'red'">
-        <div [style.height.%]="100" #parentContainer></div>
+        <div [style.height.%]="100" [style.position]="'relative'" #parentContainer></div>
```
why are we using variable binding syntax with static values here? Instead of doing this you can create classes in the scss

luisb90 replied:
none of the pages have scss files but you right, doesnt mean cant take 15 secs to add one ;) added and moved styles

aacebo commented:
If you go to your in-component demo and open the drawer, then click 'open details', the new global drawer opens but a backdrop is over the entire screen that renders all other elements unclickable.

luisb90 commented:
> If you go to your in-component demo and open the drawer, then click 'open details', the new global drawer opens but a backdrop is over the entire screen that renders all other elements unclickable.

Fixed, sneaky typo in the host bindings

luisb90 commented:
> Getting the following error when I trigger a notification on the notifications demo page:
> 
> ![image](https://user-images.githubusercontent.com/8687/72822069-c4444180-3c71-11ea-949c-1a2c77435469.png)
> 
> Assuming caused by the change in the injection service

updated and fixed, all dialog components should be working as expected
